ksqldb-rest-client/src/main/java/io/confluent/ksql/rest/client/KsqlClient.java

```diff
@@ -55,16 +58,33 @@ public KsqlClient(
     this.basicAuthHeader = createBasicAuthHeader(
         Objects.requireNonNull(credentials, "credentials"));
     this.localProperties = Objects.requireNonNull(localProperties, "localProperties");
+    this.socketAddressFactory = SocketAddress::inetSocketAddress;
     this.httpNonTlsClient = createHttpClient(vertx, clientProps, httpClientOptions, false);
     this.httpTlsClient = createHttpClient(vertx, clientProps, httpClientOptions, true);
   }
 
+  public KsqlClient(
```

Comment:
Worth adding java docs on the params?  The new `socketAddressFactory` probably isn't obvious